src/skagent/models/benchmarks.py

```diff
@@ -0,0 +1,1048 @@
+        "block": d4_block,
+        "calibration": d4_calibration,
+        "analytical_policy": d4_analytical_policy,
+    },
+    "U-1": {
+        "block": u1_block,
+        "calibration": u1_calibration,
+        "analytical_policy": u1_analytical_policy,
+    },
+    "U-2": {
+        "block": u2_block,
+        "calibration": u2_calibration,
+        "analytical_policy": u2_analytical_policy,
+    },
+    "U-3": {
+        "block": u3_block,
+        "calibration": u3_calibration,
+        "analytical_policy": u3_analytical_policy,
+    },
+    "U-4": {
+        "block": u4_block,
+        "calibration": u4_calibration,
+        "analytical_policy": u4_analytical_policy,
+    },
+    "U-5": {
+        "block": u5_block,
+        "calibration": u5_calibration,
+        "analytical_policy": u5_analytical_policy,
+    },
+    "U-6": {
+        "block": u6_block,
+        "calibration": u6_calibration,
+        "analytical_policy": u6_analytical_policy,
+    },
+}
+
+
+def get_benchmark_model(model_id: str) -> DBlock:
+    """Get benchmark model by ID (D-1, D-2, D-3, D-4, U-1, U-2, U-3, U-4, U-5, U-6)"""
+    if model_id not in BENCHMARK_MODELS:
+        available = list(BENCHMARK_MODELS.keys())
+        raise ValueError(f"Unknown model '{model_id}'. Available: {available}")
+
+    return BENCHMARK_MODELS[model_id]["block"]
+
+
+def get_benchmark_calibration(model_id: str) -> Dict[str, Any]:
+    """Get benchmark calibration by model ID"""
+    if model_id not in BENCHMARK_MODELS:
+        available = list(BENCHMARK_MODELS.keys())
+        raise ValueError(f"Unknown model '{model_id}'. Available: {available}")
+
+    return BENCHMARK_MODELS[model_id]["calibration"].copy()
+
+
+def get_analytical_policy(model_id: str) -> Callable:
+    """Get analytical policy function by model ID"""
+    if model_id not in BENCHMARK_MODELS:
+        available = list(BENCHMARK_MODELS.keys())
+        raise ValueError(f"Unknown model '{model_id}'. Available: {available}")
+
+    calibration = BENCHMARK_MODELS[model_id]["calibration"]
+    policy_func = BENCHMARK_MODELS[model_id]["analytical_policy"]
+
+    return policy_func(calibration)
+
+
+def list_benchmark_models() -> Dict[str, str]:
+    """List all 10 analytically solvable discrete-time models from the catalogue"""
+    return {
+        model_id: model_info["calibration"]["description"]
+        for model_id, model_info in BENCHMARK_MODELS.items()
+    }
+
+
+def validate_analytical_solution(
+    model_id: str, test_points: int = 10, tolerance: float = EPS_VALIDATION
+) -> Dict[str, Any]:
+    """Validate analytical solution satisfies optimality conditions and budget constraints"""
+
+    if model_id not in BENCHMARK_MODELS:
+        return {"error": f"Unknown model: {model_id}"}
+
+    try:
+        calibration = get_benchmark_calibration(model_id)
+        analytical_policy = get_analytical_policy(model_id)
+
+        # Generate appropriate test states for each model
+        if model_id == "D-1":
```

Comment:
It's weird to have this general method to have so much that is hard-coded about specific benchmark models. All other information (block, calibration, analytical policy) for each model is handled separately and stored in a dictionary. This makes it easily extensible. This method is _not_ easy to extend, because it involves changing hard-coded stuff in a method.

Is there a way to refactor this so that it refers to a different more general data structure for the test states?